src/components/JsonEditorContainer/JsonEditorContainer.tsx

```diff
@@ -18,10 +23,18 @@ const createValidInsertMenu = (submenu: MenuItem[] | undefined, validInsertItems
             let matchingItemCountWithSameDesc = 0;
 
             Object.keys(validInsertItems).forEach((key: any) => {
-                if (!existingKeys.includes(key) &&
-                    subItem.text === key &&
+                if (subItem.text === key &&
                     subItem.title === validInsertItems[key].description) {
-                    validMenuItems.push(subItem);
+                    /**
+                     * filter alredy added items from insert menu
+                     * unless it's map
+                     */
+                    if (!(resultNode instanceof AdditionalNode) && !existingKeys.includes(key)) {
+                        validMenuItems.push(subItem);
+                    }
+                    if(resultNode instanceof AdditionalNode) {
```

Comment:
Can we handle line:32 by the else part of this condition. And check existingKeys.includes(key) at the top?